statsmodels/survey/survey_table.py

```diff
@@ -0,0 +1,188 @@
+from __future__ import division
+import numpy as np
+import pandas as pd
+from statsmodels.survey import summary_stats as ss
+
+
+class SurveyTable(object):
+    # assumes data is nx2 ie the user specified which two
+    # columns to work with
+    def __init__(self, design, data):
+        self.design = design
+        self._m, self._p = data.shape
+        self.df = pd.concat([pd.DataFrame(data),
+                            pd.DataFrame(self.design.weights)], axis=1)
+
+        # if just given one column
+        if self.df.shape[1] == 2:
+            self.df.columns = ['var1', 'weights']
+            self.df_group = self.df.groupby(['var1'])
+            self.table = self.df_group.weights.sum().unstack()
+            self.table = self.table.fillna(0)
+        elif self.df.shape[1] == 3:
+            self.df.columns = ["var1", "var2", "weights"]
+            self.df_group = self.df.groupby(['var1', 'var2'])
+            # the 'total' for each group
+            self.table = self.df_group.weights.sum().unstack()
+            self.table = self.table.fillna(0)
+
+        else:
+            return ValueError("data should only have 1 or 2 columns")
+
+        self._row_sum = self.table.sum(axis=1)  # shape is (var1_unique_labs, )
+        self._col_sum = self.table.sum(axis=0)  # shape is (var2_unique_labs, )
+        self._row_prop = self.table.div(self._row_sum, axis=0)
+        self._col_prop = self.table.div(self._col_sum, axis=1)
+        self._tot_sum = self.table.sum().sum()
+        self._cell_prop = self.table / self._tot_sum
+        self._row_marginal = self._row_sum / self._tot_sum
+        self._col_marginal = self._col_sum / self._tot_sum
+        # estimated proportion under the null hypothesis of independence
+        self._null = np.outer(self._row_marginal, self._col_marginal)
+
+    def __str__(self):
+        tab = self._cell_prop.copy()
+        # tab['col_tot'] = tab.sum(axis=1)
+        # tab.loc['row_tot'] = tab.sum(axis=0)
+        print(tab)
+        return 'key: cell_proportions'
+
+    def test_pearson(self, cell_prop=True):
+        cell_diff_square = np.square((self._cell_prop - self._null))
+        # uncorrected stat
+        self.pearson = self._m * (cell_diff_square / self._null).sum().sum()
+        # rao and scott correction
+        self._delta(cell_prop)
+        if self._delta_est.ndim == 1:
+            self._trace = self.delta_est
+            self._trace_sq = np.square(self._delta_est)
+        else:
+            self._trace = np.trace(self._delta_est)
+            self._trace_sq = np.trace(np.square(self._delta_est))
+
+        self.pearson_chi = self.pearson * (self._trace / self._trace_sq)
+        self.dof_chi = np.square(self._trace) / self._trace_sq
+        self.pearson_f = self.pearson / self._trace
+        v = self.design.n_clust - self.design.n_strat
+        self.dof_F = (self.dof_chi, v * self.dof_chi)
+
+    def test_lrt(self, cell_prop=True):
```

Comment:
Should return a result

Reply:
You can use this `_Bunch`  class to return results without needing to define a custom results class:

https://github.com/statsmodels/statsmodels/blob/master/statsmodels/stats/contingency_tables.py#L59

Reply:
See here for an example of usage:

https://github.com/statsmodels/statsmodels/blob/d0af408e00032e5ca5d325629409fb8be80f8eca/statsmodels/stats/contingency_tables.py#L204